### ISSUE TITLE
Why do we need to name the exported function? (somewhat confusing to a newbie)

### ISSUE THREAD

**Issue by simensen** [erikras/ducks-modular-redux#21]
Rule one sent me down a weird little rabbit hole that I might not have been down if I were a little more experienced with this whole `import`/`export` business. :) Still, I think there might be room for clarity here.

> MUST `export default` a function called `reducer()`

... when paired with the example:

```javascript
export default function reducer(state = {}, action = {}) {
  switch (action.type) {
    // do reducer stuff
    default: return state;
  }
}
```

... made me think that since the function was named `reducer` that I'd be able to do something like:

```javascript
import { reducer as personReducer } from 'redux/modules/person'
```

This is sadly not the case at all, at least for my current babel/es6 config (maybe it works for AMD/CommonJS/whatever-else there is out there?).

If I go just based off of rule itself, it makes me think that I need to export a function named `reducer` that is also the default export.

```javascript
export function reducer(state = {}, action = {}) {
  switch (action.type) {
    // do reducer stuff
    default: return state;
  }
}

export default reducer
```

This bit makes me think that the intention is not actually to export a function named `reducer`:

> ...and it will only import the action creators, ready to be passed to `bindActionCreators()`.

So I'm wondering why rule one says anything about the name of the function (`reducer`) and why the actual code example bothers to name the function `reducer` if it is never used anywhere and is not intended to be used anywhere.

---

I've still got a lot to learn and I'm guessing this has to do with different versions of javascript being able to have non-named functions or something and standardizing on the name just makes sense so the same module can be used anywhere. But this particular oddity contributed to me losing at least an hour's worth of work tonight because I couldn't figure out why `import { reducer as someOtherName } from 'my/duck'` was giving me `undefined`. :)

If there are no good reasons for the function being named reducer, I think it would have been more clear to me if the rule had been:

> MUST `export default` a function

... with the example:

```javascript
export default function(state = {}, action = {}) {
  switch (action.type) {
    // do reducer stuff
    default: return state;
  }
}
```

**Comment by erikras** [comment]
I don't think you need to for a technical reason, but it's something ESLint suggests, so I normally do. It helps to self-document the code.

**Comment by hnordt** [comment]
Another benefit, if reducer() throws the error will contain the function name (reducer).

**Comment by aikeru** [comment]
@simensen 
The reason
```
import { reducer as personReducer } from 'redux/modules/person'
```
doesn't work is because "reducer" is the default export, so to consumers it isn't named (just as you can "default export" a variable, which may have a name internal to the file/module).
Since we know it's the default, you can do this:
```
import { default as personReducer } from 'redux/modules/person'
```
